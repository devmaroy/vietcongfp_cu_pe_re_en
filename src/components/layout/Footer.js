import React from 'react';

const Footer = () => {
    const iconsCopyright = (
        <div>
            Icons made by  
            <a href="https://www.freepik.com/" title="Freepik"> Freepik </a> 
             from <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com </a> 
             is licensed by 
            <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank"> CC 3.0 BY </a>
        </div>
    );

    return (
        <div className="content-container">
            <div className="footer">
                <p>Copyright &copy; { new Date().getFullYear() }</p>
                <p>All rights reserved.</p>
                <p>Made with <i className="heart-symbol"></i> by <a href="https://marekmatejovic.com/">devmaroy</a></p> 
                { iconsCopyright }
            </div>
        </div>
    );
};

export default Footer;