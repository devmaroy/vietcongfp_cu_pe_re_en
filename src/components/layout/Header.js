import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../../actions/auth';

export class Header extends React.Component {
    state = {
        isOpen: false
    }

    showMenu = ( e ) => {
        this.setState( ( prevState ) => ( { isOpen: !prevState.isOpen } ) );
    }

    render() {
        const { startLogout, isAuthenticated, user } = this.props;

        return (
            <div className="content-container">
                <header className="header">
                    <div className="header-content">
                        <button onClick={ this.showMenu } className="hamburger-menu">
                            <span className={ classnames( 'hamburger-menu__icon', { 'is-open': this.state.isOpen } ) }></span>
                        </button> 
            
                        <nav className="header-nav">
                            <ul className={ classnames( 'header-nav__list list-unstyled', { 'is-open': this.state.isOpen } ) }>
                                <li className="header-nav__list-item">
                                    <NavLink
                                        to="/"
                                        className="header-nav__item"
                                        activeClassName="header-nav__item--active"
                                        exact={ true }
                                    >
                                        Homepage
                                    </NavLink>
                                </li>
                                <li className="header-nav__list-item">
                                    <NavLink
                                        to="/about"
                                        className="header-nav__item"
                                        activeClassName="header-nav__item--active"
                                    >
                                        About vietcong
                                    </NavLink>
                                </li>
                                <li className="header-nav__list-item">
                                    <NavLink
                                        to="/clans"
                                        className="header-nav__item"
                                        activeClassName="header-nav__item--active"
                                    >
                                        Clans
                                    </NavLink>
                                </li>
                            </ul>
                        </nav>
                    </div>                   
                    {
                        isAuthenticated && (
                            <div className="header-meta">
                                <div className="header-controls">
                                    <h3 className="header-controls__title">{ user.name }</h3>
                                    <Link
                                        to="/posts/create"
                                        className="button add-post"
                                    >
                                        Add post
                                    </Link>
                                    <Link
                                        to="/clans/create"
                                        className="button add-clan"
                                    >
                                        Add clan
                                    </Link>
                                    <button 
                                        className="button button--link logout-link"
                                        onClick={ startLogout }
                                    >
                                        Logout
                                    </button>
                                </div>
                            </div>
                        )
                    } 
                </header>
            </div>
        );
    }
}

Header.propTypes = {
    isAuthenticated: PropTypes.bool,
    user: PropTypes.object,
    startLogout: PropTypes.func.isRequired
};

const mapStateToProps = ( state ) => ({
    isAuthenticated: state.auth.isAuthenticated,
    user: state.auth.user
});

const mapDispatchToProps = ( dispatch ) => ({
    startLogout: () => dispatch( startLogout() )
});

export default connect( mapStateToProps, mapDispatchToProps )( Header );