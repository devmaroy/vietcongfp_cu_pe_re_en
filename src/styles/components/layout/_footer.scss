/*------------------------------------*\
    # Footer
\*------------------------------------*/


/* Footer */

.footer {
    text-align: center;
    color: $off-white;
    margin: $xl-size 0;
}