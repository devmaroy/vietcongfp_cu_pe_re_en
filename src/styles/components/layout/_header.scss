/*------------------------------------*\
    # Header
\*------------------------------------*/


/* Header */

.header {
    background: $green;
    border-bottom: .2rem solid $off-green;
    margin-top: $xl-size;
    padding: 0 $m-size;

    @media ( min-width: $desktop-breakpoint ) {
        display: flex;
        justify-content: space-between;
    }
}


/* Header Content */

.header-content {
    text-align: center;

    @media ( min-width: $desktop-breakpoint ) {
        display: flex;
        align-items: center;
        padding-top: 0;
    }
}


/* Header Meta */

.header-meta {
    padding-bottom: $l-size;
}


/* Header Controls */

.header-controls__title {
    color: $off-white;
    margin-bottom: $l-size;
}

.header-controls {
    text-align: center;

    @media ( min-width: $desktop-breakpoint ) {
        text-align: left;
    }
}