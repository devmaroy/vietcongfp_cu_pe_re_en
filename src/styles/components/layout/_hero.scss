/*------------------------------------*\
    # Hero
\*------------------------------------*/


.hero {
    background: url( '/images/backgroundtwo.jpg' );
    background-size: cover;
    border: .3rem solid #505250;
    border-radius: .3rem;
    height: 30rem;
    margin-top: $m-size;
    

    display: flex;
    justify-content: flex-end;
}

.hero-content {
    max-width: 55rem;
    margin: 0 $m-size;
}

.hero-content__overlay {
    background: rgba( black, .7 );
    padding: $m-size;
    color: $off-white;
    text-align: center;
}

.hero-content__title {
    text-transform: uppercase;
}