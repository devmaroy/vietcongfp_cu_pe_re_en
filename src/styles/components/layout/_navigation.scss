/*------------------------------------*\
    # Navigation
\*------------------------------------*/


/* Hamburger Menu */

.hamburger-menu {
    display: block;
    background: none;
    outline: none;
    border: none;

    padding: $l-size;
    width: 10rem;
    margin-left: auto;
    

    cursor: pointer;

    @media ( min-width: $desktop-breakpoint ) {
        display: none;
    }
}

.hamburger-menu__icon,
.hamburger-menu__icon::before,
.hamburger-menu__icon::after {
    content: '';
    display: block;
    width: 4rem;
    height: .3rem;
    background: $off-white;

    transition: $transition;
}

.hamburger-menu__icon::before {
    transform: translateY( -9px );
}

.hamburger-menu__icon::after {
	transform: translateY( 6px );
}

.hamburger-menu__icon.is-open {
	transform: rotate( 45deg );

	&::after {
		transform: rotate( -90deg ) translateX( 3px );
	}

	&::before {
		opacity: 0;
	}
}


/* Header Navigation */

.header-nav__list {
    display: none;
    padding-top: $m-size;
    margin-bottom: $xl-size;

    &.is-open {
        display: block;
    }

    @media ( min-width: $desktop-breakpoint ) {
        display: block;
        padding-top: 0;
        margin-bottom: 0;
    }
}

.header-nav__list-item {
    margin-bottom: $m-size;

    &:last-child {
        margin-bottom: 0;
    }

    @media ( min-width: $desktop-breakpoint ) {
        display: inline-block;
        margin-bottom: 0;
        padding-right: $s-size;
    }
}

.header-nav__item {
    color: $light-green;
    display: block;
    text-decoration: none; 
    text-transform: uppercase;
    padding: $l-size 0;
    line-height: 1;
    
    transition: $transition;

    &:focus,
    &:hover {
        color: $off-white;
    }
}

.header-nav__item--active {
    color: $off-white;
}