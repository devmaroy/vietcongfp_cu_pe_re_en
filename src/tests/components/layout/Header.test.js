import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../../components/layout/Header';

const user = {
    name: 'Josh'
};

test( 'should render Header correctly for no login user', () => {
    const wrapper = shallow( 
            <Header 
                startLogout={ () => {} } 
                isAuthenticated={ false } 
                user={ {} } 
            /> 
        );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should render Header correctly for logged in user', () => {
    const wrapper = shallow( 
            <Header 
                startLogout={ () => {} } 
                isAuthenticated={ true } 
                user={ user } 
            /> 
        );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should call startLogout on button click ', () => {
    const startLogout = jest.fn();

    const wrapper = shallow( 
            <Header 
                startLogout={ startLogout } 
                isAuthenticated={ true } 
                user={ user } 
            /> 
        );

    wrapper.find( '.logout-link' ).simulate( 'click' );

    expect( startLogout ).toHaveBeenCalled();
});

test( 'should show header-meta if authenticated', () => {
    const startLogout = jest.fn();

    const wrapper = shallow( 
            <Header 
                startLogout={ startLogout } 
                isAuthenticated={ true } 
                user={ user } 
            /> 
        );

    expect( wrapper.find( '.header-meta' ).exists() ).toBeTruthy();
});

test( 'should not show header-meta if not authenticated', () => {
    const startLogout = jest.fn();

    const wrapper = shallow( 
            <Header 
                startLogout={ startLogout } 
                isAuthenticated={ false } 
                user={ {} } 
            /> 
        );

    expect( wrapper.find( '.header-meta' ).exists() ).toBeFalsy();
});