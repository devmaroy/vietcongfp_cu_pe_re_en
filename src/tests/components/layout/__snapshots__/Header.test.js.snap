// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly for logged in user 1`] = `
<div
  className="content-container"
>
  <header
    className="header"
  >
    <div
      className="header-content"
    >
      <button
        className="hamburger-menu"
        onClick={[Function]}
      >
        <span
          className="hamburger-menu__icon"
        />
      </button>
      <nav
        className="header-nav"
      >
        <ul
          className="header-nav__list list-unstyled"
        >
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              exact={true}
              to="/"
            >
              Homepage
            </NavLink>
          </li>
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              to="/about"
            >
              About vietcong
            </NavLink>
          </li>
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              to="/clans"
            >
              Clans
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="header-meta"
    >
      <div
        className="header-controls"
      >
        <h3
          className="header-controls__title"
        >
          Josh
        </h3>
        <Link
          className="button add-post"
          replace={false}
          to="/posts/create"
        >
          Add post
        </Link>
        <Link
          className="button add-clan"
          replace={false}
          to="/clans/create"
        >
          Add clan
        </Link>
        <button
          className="button button--link logout-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`should render Header correctly for no login user 1`] = `
<div
  className="content-container"
>
  <header
    className="header"
  >
    <div
      className="header-content"
    >
      <button
        className="hamburger-menu"
        onClick={[Function]}
      >
        <span
          className="hamburger-menu__icon"
        />
      </button>
      <nav
        className="header-nav"
      >
        <ul
          className="header-nav__list list-unstyled"
        >
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              exact={true}
              to="/"
            >
              Homepage
            </NavLink>
          </li>
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              to="/about"
            >
              About vietcong
            </NavLink>
          </li>
          <li
            className="header-nav__list-item"
          >
            <NavLink
              activeClassName="header-nav__item--active"
              className="header-nav__item"
              to="/clans"
            >
              Clans
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
